packages/esm-appointments-app/src/form/appointments-form.component.tsx

```diff
@@ -4,34 +4,33 @@ import dayjs from 'dayjs';
 import {
   Button,
   ButtonSet,
-  DatePickerInput,
   DatePicker,
+  DatePickerInput,
   Form,
   InlineLoading,
   MultiSelect,
   NumberInput,
+  RadioButton,
+  RadioButtonGroup,
   Select,
   SelectItem,
   Stack,
-  RadioButtonGroup,
-  RadioButton,
   TextArea,
-  TimePickerSelect,
   TimePicker,
+  TimePickerSelect,
   Toggle,
 } from '@carbon/react';
 import { Controller, useForm } from 'react-hook-form';
 import { zodResolver } from '@hookform/resolvers/zod';
 import { z } from 'zod';
 import {
-  useLocations,
-  useSession,
+  ResponsiveWrapper,
   showSnackbar,
-  useLayoutType,
   useConfig,
-  ResponsiveWrapper,
+  useLayoutType,
+  useLocations,
+  useSession,
 } from '@openmrs/esm-framework';
-import { convertTime12to24 } from '@openmrs/esm-patient-common-lib';
```

Comment:
The use of this function from patient-common-lib was a big part of the problem, as it is what is responsible for (silently) returning a value of "midnight" if the converted time fails to match the expected format.  This gets rid of the usage of this and adds the appropriate handling directly.

Reply:
@ibacher @denniskigen  assumedly we should ticket and fix the convertTime12to24 as well, unless for some reason that behavior is expected?